merkle/ops.go

```diff
@@ -114,9 +123,13 @@ func (mt *merkleTreeImp) Print() string {
 	if mt.root == nil {
 		buffer.WriteString("Merkle Tree: Empty tree.\n")
 	} else {
-
-		buffer.WriteString(fmt.Sprintf("Merkle tree: root hash <%s>\n", hex.EncodeToString(mt.GetRootHash())[:6]))
-		buffer.WriteString(mt.root.print(mt.treeData, mt.userData))
+		rootHash, err := mt.GetRootHash()
+		if err == nil {
```

Comment:
Errors now get rendered on the print. I am not sure if we want this but it can be really useful.

Reply:
I think it is fine in the print - should be fairly exceptional circumstances as hashing will only fails if marshaling the protobuf data fails.